File: react frontend/src/services/FootballClubService.js
import axios from "axios";

const FOOTBALLCLUB_API_BASE_URL = "http://localhost:8080/api/v1/footballClubs";

class FootballClubService {
  getAllFootballClub() {
    return axios.get(FOOTBALLCLUB_API_BASE_URL);
  }

  createFootballClub(footballClub) {
    return axios.post(FOOTBALLCLUB_API_BASE_URL, footballClub);
  }

  getFootballClubById(footballClubId) {
    return axios.get(FOOTBALLCLUB_API_BASE_URL + "/" + footballClubId);
  }

  updateFootballClub(footballClub, footballClubId) {
    return axios.put(FOOTBALLCLUB_API_BASE_URL + "/" + footballClubId, footballClub);
  }
  deleteFootballClub(footballClubId) {
    return axios.delete(FOOTBALLCLUB_API_BASE_URL + "/" + footballClubId);
  }
}

export default new FootballClubService();

File: springboot-backend/src/main/java/com/league/service/FootballClubService.java
package com.league.service;

import com.league.FootballClub;

import java.util.List;


public interface FootballClubService {

    List<FootballClub> getAllFootballClub();

    FootballClub createFootballClub(FootballClub footballClub);

    FootballClub getFootballClubById(int footballClubId);

    FootballClub updateFootballClub(int footballClubId,FootballClub footballClub);

    void deleteFootballClub(int footballClubId);

}
